### ISSUE TITLE
Role doesn't appear to bootstrap from 'ground up'

### ISSUE THREAD

**Issue by goetzk** [tohuwabohu/puppet-drupal#3]
Along with the changes in PR #2 I've hit what appears to be a blocker.
```
Error: /usr/local/sbin/drupal-update.sh /opt/drupal.org/www.lan-f90ede3c829ae2d54b7e87455b735ed9db671160 www.lan >> /var/log/drush/www.lan.log 2>&1 returned 1 instead of one of [0]
Error: /Stage[main]/Main/Node[au.lan]/Drupal::Site[www.lan]/Exec[update-drupal-www.lan-database]/returns: change from notrun to 0 failed: /usr/local/sbin/drupal-update.sh /opt/drupal.org/www.lan-f90ede3c829ae2d54b7e87455b735ed9db671160 www.lan >> /var/log/drush/www.lan.log 2>&1 returned 1 instead of one of [0]
```

The part where it fails (from the log) is
```
        Command variable-set needs a higher bootstrap level to run - you will need to invoke drush from a more functional Drupal environment to run this command.                                   [error]
The drush command 'vset maintenance_mode 1' could not be executed.                                                                                                                          [error]
Drush was not able to start (bootstrap) the Drupal database.                                                                                                                                [error]
Hint: This may occur when Drush is trying to:
 * bootstrap a site that has not been installed or does not have a configured database. In this case you can select another site with a working database setup by specifying the URI to use
with the --uri parameter on the command line. See `drush topic docs-aliases` for details.
 * connect the database through a socket. The socket file may be wrong or the php-cli may have no access to it in a jailed shell. See http://drupal.org/node/1428638 for details.
```

Has this role been tested doing deployments from bare metal or is an existing install a requirement?

I'm trying to deploy using puppet 3.7 onto Centos 7.
```
Drush was attempting to connect to: 
 Drupal version         :  7.37                                                                 
 Site URI               :  www.lan                                                          
 Default theme          :  garland                                                              
 Administration theme   :  garland                                                              
 PHP executable         :  /usr/bin/php                                                         
 PHP configuration      :  /etc/php.ini                                                         
 PHP OS                 :  Linux                                                                
 Drush version          :  6.6.0                                                                
 Drush configuration    :                                                                       
 Drush alias files      :  /etc/drush/aliases.drushrc.php /etc/drush/au.aliases.drushrc.php   
 Drupal root            :  /opt/drupal.org/www.lan-f90ede3c829ae2d54b7e87455b735ed9db671160 
 Site path              :  sites/default                                                        
 File directory path    :  sites/default/files    
```

**Comment by tohuwabohu** [comment]
So far the module has been only used on existing installations. But if it could work from scratch, that would be great for the acceptance tests. A while ago, I've tried to make drush work to create an example site upfront but failed somehow.

What do you reckon should have happened (assuming the Puppet run would finish without error): you'd see the installation wizard?

**Comment by goetzk** [comment]
yeah, i'd expect the install wizard or a configured system if settings.php was provided. If the role is only for rebuilding existing instances it would be good if the realm/forge description mentioned that - at the moment its unclear and sounds like a clean install can work.

**Comment by tohuwabohu** [comment]
Fixed in version 1.2.1.
